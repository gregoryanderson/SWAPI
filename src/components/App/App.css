@font-face {font-family: "STARWARS";
  src: url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.eot");
  src: url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.eot?#iefix") format("embedded-opentype"),
  url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.woff2") format("woff2"),
  url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.woff") format("woff"),
  url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.ttf") format("truetype"),
  url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.svg#STARWARS") format("svg");
}

* {
  box-sizing: border-box;
  font-family: 'STARWARS';
  color: #F9E60A;
  -webkit-text-stroke-width: .1px; 
  -webkit-text-stroke-color: black; 
}

html {
  background-image: url('http://static.hdw.eweb4.com/media/wallpapers_dl/1/111/1101575-lightsaber.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  background-attachment: fixed;
  height: 1000px;
}

/* nav bar */
.nav__header {
  list-style-type: none;
  justify-content: space-around;
  align-items: center;
  display: inline-flex;
  width: 100%;
}

.nav__h1 {
  font-size: 50px;
  margin: 30px 0px 30px 40px;
  /* width: 100%; */
  min-width: 600px;
}

.nav__section--links {
  width: 100%;
  display: flex;
  justify-content: space-around;
  align-items: center;
  /* text-align: center; */
}

.selected{
  color:#FF0200
}

.nav__link {
  display: block;
  padding-top: 2px;
}

.nav__div--favorites {
  display: flex;
  justify-content: space-around;
  align-items: center;
}

/* home header */
.home__h1{
  text-align: center;
  width: 500px;
  display: flex;
  justify-content: center;
  margin: auto;
  margin-top: 50px;
}

.favorite__link {
  padding-right: 7px;
} 

/* card container */
.app__section {
  padding: 20px;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 20px;
}

/* selected card */
.selected__section {
  display: flex;
  justify-content: center;
}

.loading{
  background-size: cover;
  height: 100%;
  width: 100%
}

/* //media queries */
/* .nav__header{
  display: inline-flex;
} */

/* this is the symbol font family for if/when we wish to use it */

/* @font-face {font-family: "StarWars  Kit";
  src: url("//db.onlinewebfonts.com/t/a5522bbd15c8d4afaf762a74906cebf3.eot");
  src: url("//db.onlinewebfonts.com/t/a5522bbd15c8d4afaf762a74906cebf3.eot?#iefix") format("embedded-opentype"),
  url("//db.onlinewebfonts.com/t/a5522bbd15c8d4afaf762a74906cebf3.woff2") format("woff2"),
  url("//db.onlinewebfonts.com/t/a5522bbd15c8d4afaf762a74906cebf3.woff") format("woff"),
  url("//db.onlinewebfonts.com/t/a5522bbd15c8d4afaf762a74906cebf3.ttf") format("truetype"),
  url("//db.onlinewebfonts.com/t/a5522bbd15c8d4afaf762a74906cebf3.svg#StarWars  Kit") format("svg");
} */
