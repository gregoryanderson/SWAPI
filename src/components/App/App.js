import React, { Component }from 'react';
import People from '../People/People'
import Planets from '../Planets/Planets'
import Vehicles from '../Vehicles/Vehicles'
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      info: []
    }
  }

  fetchWorld = (peopleInfo) => {
    console.log(peopleInfo)
    const promises = peopleInfo.map(person => {
      return fetch (person.homeworld)
        .then(response => response.json())
        .then(data => ({ ...data, personName: person.name, personSpecies: person.species}))
        .then(personWorldData => this.fetchSpeciesInfo(personWorldData))
        .then(ish => console.log('boo', ish))
        .catch(error => console.log(error))
      })
      return Promise.all(promises)
    }
    
    fetchSpeciesInfo = (personWorldInfo) => {
      console.log('in the damn species method', personWorldInfo)
      const promises = personWorldInfo.personSpecies.map(species => {
        fetch(personWorldInfo.personSpecies[0])
        .then(response => response.json())
        .then(data => ({...data, personName: personWorldInfo.personName, personSpecies: data.name, personPlanet: personWorldInfo.name, personPlanetPopulation: personWorldInfo.population, personLanguage: data.language}))
        .catch(error => console.log(error))
    })
    Promise.all(promises)
  }


  componentDidMount() {
    fetch('https://swapi.co/api/people/')
      .then(response => response.json())
      .then(data => this.fetchWorld(data.results))
      .then(info => this.setState({info}))
      .catch(error => console.log(error))
  }

  render() {
    return (
    <div>
      <People />
      <Planets />
      <Vehicles />
    </div>
    )
  }
}

export default App;
