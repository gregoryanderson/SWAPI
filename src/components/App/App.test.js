import React from "react";
import { shallow } from "enzyme";
import App from "./App";

describe("App", () => {
  it("should match the snapshot", () => {
    const wrapper = <App />;
    expect(wrapper).toMatchSnapshot();
  });
});
