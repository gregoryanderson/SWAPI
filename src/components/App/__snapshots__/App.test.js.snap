// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `<App />`;

exports[`renders without crashing 1`] = `<App />`;

exports[`should match the snapshot 1`] = `<App />`;
